subsys/net/ip/l2/arp.c

```diff
@@ -8,7 +8,7 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-#if defined(CONFIG_NET_DEBUG_ARP)
+#if defined(CONFIG_NET_DEBUG_ARP) || defined(CONFIG_NET_LOG_GLOBAL)
```

Comment:
This is problematic if you want to check if debugging is activated in the code.
So if the code has now
```
if (IS_ENABLED(CONFIG_NET_DEBUG_ARP)) {
...
}
```
then one would need to have
```
if (IS_ENABLED(CONFIG_NET_DEBUG_ARP) || IS_ENABLED(CONFIG_NET_LOG_GLOBAL)) {
...
}
```

IMHO, doing it like this is not really needed as we can automatically enable ARP debugging in Kconfig file when necessary. We just need to add needed debug options there.

Reply:
> This is problematic if you want to check if debugging is activated in the code.

How often is that needed? If you look at that code, it has

~~~
#if defined(CONFIG_NET_DEBUG_ARP) || defined(CONFIG_NET_LOG_GLOBAL)
#define NET_LOG_ENABLED 1
#endif
~~~

So, you can use NET_LOG_ENABLED afterwards.